unstdlib/standard/list_.py

```diff
@@ -145,6 +145,32 @@ def iterate_flatten(q):
 
     return chain.from_iterable(q)
 
+def partition(iterable, predicate=bool):
```

Comment:
You've got a point, although in some cases the ```tee``` copy of the original iterator is more efficient than creating a new iterator. My bigger point for this function is actually readability/clarity of the code using it. Consider the example above written as two ifilter statements, you would have to duplicate the predicate.